### ISSUE TITLE
Unable to Reproduce Paper Results Using Provided Model

### ISSUE THREAD

**Issue by JoJo0217** [kuleshov-group/mdlm#29]
Hi, thank you for releasing the code and models.
I’ve been trying to reproduce the results reported in the paper using the provided MDLM checkpoint and ppl_eval mode, but I'm running into some inconsistencies.

I downloaded the MDLM model from the provided Google Drive link and ran the following command for evaluation:

```bash
CUDA_VISIBLE_DEVICES=7 python main.py \
  mode=ppl_eval \
  loader.batch_size=16 \
  loader.eval_batch_size=16 \
  data=openwebtext-split \
  model=small \
  sampling.predictor=ddpm \
  sampling.steps=1024 \
  parameterization=subs \
  backbone=dit \
  model.length=1024 \
  sampling.num_sample_batches=0 \
  eval.checkpoint_path=/share/mdlm_origin_ckpt/mdlm.ckpt \
  +wandb.offline=true
```
My results:
|Model | LM1B | LAMBADA | PTB | AG News | Pubmed | Arxiv | Wikitext|
|-- | -- | -- | -- | -- | -- | -- | --|
|AR | 55.55 | 48.44 | 105.81 | 59.14 | 48.77 | 41.23 | 34.17|
|MDLM | 67.94 | 48.40 | 83.82 | 64.79 | 42.87 | 37.73 | 36.90|

Paper results:
|Model | LM1B | LAMBADA | PTB | AG News | Pubmed | Arxiv | Wikitext|
|-- | -- | -- | -- | -- | -- | -- | --|
|AR | 51.25 | 51.28 | 82.05 | 52.09 | 49.01 | 41.73 | 41.73|
|MDLM | 67.01 | 47.52 | 95.26 | 61.15 | 41.89 | 37.37 | 32.83|

My observations:
- I expected differences in MDLM results due to the stochastic nature of the generation process (e.g., t can vary based on GPU count or batch size, even with the same seed), especially on smaller datasets.

- However, I also see discrepancies in the AR model results, which shouldn't be affected by those factors.

This suggests there may be something I misunderstood or missed in the evaluation process.

Are there any specific settings (batch size, hardware setup, data splits, etc.) I might have deviated from compared to the paper's experiments?
Is there a fixed seed or other configuration that must be enforced to align with the reported results?
Could there be post-processing or tokenization differences that explain the gap?

Any insight or guidance would be greatly appreciated.

Thanks!

**Comment by JoJo0217** [comment]
After checking in more detail, I realized that the discrepancies were due to a mismatch in training steps between the released checkpoints and the models used in the paper.

According to the paper:

AR was trained on 524B tokens (1M steps),

MDLM and SEDD (524B tokens) were trained for 2M steps.

However, the released checkpoints are:

MDLM: 1,112,679 steps

SEDD: 985,379 steps

AR: 1,060,839 steps
